src/core/storage/compression/Decompressor.h

```diff
@@ -78,15 +85,32 @@ class DecompressorImpl {
 	unsigned long m_symbols = 0; //!< count of the file or folder m_symbols
 
 	unsigned char m_current_byte = '\0';//!< unsigned char value
-	                                    //!< that represents the current byte
+	//!< that represents the current byte
 	int m_current_bit_count = 0;        //!< integer value of current bits count
 
+	/// \brief Creating a file, when the parent directories might not exist
+	///
+	/// \param path - path to the file
+	static void create_desired_dirs(const std::string &path) {
```

Comment:
Check `std::filesystem::create_directories`.

Reply:
Yes, I know it and I use it in other parts of my compression/decompression algorithms, but here I must create a file in some directory structure whereas the parent directories of that file might not exist - so I go through them, check if they exist and if not - create the specific one.

Also check [this](https://gamedev.net/forums/topic/305454-creating-a-file-when-the-parent-directories-might-not-exist/2944881/).

Reply:
Ok, I undestand. Maybe we should consider bringing this functionality from `boost` rather then implementing it on our own. What do you thing.

PS: We also need a logging library and as far as I know, `boost` has one embedded.

Reply:
Ok, I tried it out inside CmakeList with Fetch_Content, but probably with conan will be better. Also, it is new functionality for another PR.

Reply:
Yup. I am currently trying to transition our dependency management to use `conan` so I hope we'll see that soon.